Composer/packages/lib/indexers/src/botIndexer.ts

```diff
@@ -19,7 +19,58 @@ import map from 'lodash/map';
 
 import { getLocale } from './utils/help';
 
-// Verify bot settings, files meet LUIS/QnA requirments.
+/**
+ * Check skill manifest.json.
+ * 1. Manifest should exist
+ */
+const checkManifest = (assets: BotAssets): Diagnostic[] => {
+  const { skillManifests } = assets;
+
+  const diagnostics: Diagnostic[] = [];
+  if (skillManifests.length === 0) {
+    diagnostics.push(new Diagnostic('Missing skill manifest', 'manifest.json', DiagnosticSeverity.Warning));
+  }
+  return diagnostics;
+};
+
+/**
+ * Check skill appsettings.json.
+ * 1. Missing LUIS key
+ * 2. Missing QnA Maker subscription key.
+ */
+const checkSetting = (assets: BotAssets, localStorage: { [key: string]: any }): Diagnostic[] => {
+  const { dialogs, setting } = assets;
+  const diagnostics: Diagnostic[] = [];
+
+  const useLUIS = dialogs.some((item) => !!item.luFile);
+  // if use LUIS, check LUIS authoringKey key
+  if (useLUIS) {
+    const authoringKeyFromSettings = get(setting, 'luis.authoringKey');
+    const authoringKeyFromLocal = get(localStorage, 'luis.authoringKey');
+    if (!authoringKeyFromLocal && !authoringKeyFromSettings) {
+      diagnostics.push(new Diagnostic('Missing LUIS key', 'appsettings.json', DiagnosticSeverity.Error));
```

Comment:
Are these checks run constantly?

@lei9444 We need this check to run before the selector to enable Start Bots button gets set? Is it already doing this?

Reply:
I made some update on this. When load a bot calling Indexer.index will also call BotIndexer.validate to get diagnostics
https://github.com/microsoft/BotFramework-Composer/blob/f484cdaaf4a3a7ec5356e0a7a5662ab925daf2ef/Composer/packages/client/src/recoilModel/dispatchers/utils/project.ts#L336